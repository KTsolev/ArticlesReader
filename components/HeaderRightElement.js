import React from "react";
import { View, TouchableOpacity } from "react-native";
import { Icon } from 'react-native-elements';
import { showHidePopupMenu } from '../actions/actions';
import { connect } from 'react-redux';
import MenuPopup from './MenuPopup';


class HeaderRightElement extends React.Component {
   render(){ 
       return <View style={{ marginLeft: 5, flexDirection: 'row', justifyContent: 'space-between', padding: 5, minWidth: 50}}>
            <Icon
            name='search' 
            size={20} 
            type='font-awesome'
            color='#fff' />
            <TouchableOpacity onPress={() => this.props.togglePopup(!this.props.showPopUpMenu)}>  
              <Icon 
                name='ellipsis-v' 
                size={20} 
                type='font-awesome'
                color='#fff' />
            </TouchableOpacity>
            <MenuPopup navigation={this.props.navigation}/>
        </View>
    }
}     

const mapStateToProps = (state)=> {
  return {
      showPopUpMenu: state.uiElementsState.showPopUpMenu
  }
}
  
const mapDispatchToProps = dispatch => {
    return {
      togglePopup: (toggle) => {
        dispatch(showHidePopupMenu(toggle));
      }
    }
}
  
 
export default connect(mapStateToProps, mapDispatchToProps)(HeaderRightElement);