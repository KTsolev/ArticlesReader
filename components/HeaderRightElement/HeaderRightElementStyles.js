import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: { marginLeft: 5, flexDirection: 'row', justifyContent: 'space-between',
        padding: 5, minWidth: 50 }
});