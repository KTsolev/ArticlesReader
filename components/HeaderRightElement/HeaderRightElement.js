import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';
import { showHidePopupMenu } from '../../actions/actions';
import { connect } from 'react-redux';
import MenuPopup from '../MenuPopup/MenuPopup';
import { styles } from './HeaderRightElementStyles';

class HeaderRightElement extends React.Component {
   render(){ 
       const { container } = styles;
       return <View style={container}>
            <Icon
            name='search' 
            size={20} 
            type='font-awesome'
            color='#fff' />
            <TouchableOpacity onPress={() => this.props.togglePopup(!this.props.showPopUpMenu)}>  
              <Icon 
                name='ellipsis-v' 
                size={20} 
                type='font-awesome'
                color='#fff' />
            </TouchableOpacity>
            <MenuPopup navigation={this.props.navigation}/>
        </View>
    }
}     

const mapStateToProps = (state)=> {
  return {
      showPopUpMenu: state.uiElementsState.showPopUpMenu
  }
}
  
const mapDispatchToProps = dispatch => {
    return {
      togglePopup: (toggle) => {
        dispatch(showHidePopupMenu(toggle));
      }
    }
}
  
 
export default connect(mapStateToProps, mapDispatchToProps)(HeaderRightElement);