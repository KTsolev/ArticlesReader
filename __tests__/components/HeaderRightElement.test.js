import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import { TouchableOpacity } from 'react-native'
import { render, fireEvent } from 'react-native-testing-library';
import configureMockStore from 'redux-mock-store';
import HeaderRightElement from '../../components/HeaderRightElement/HeaderRightElement';
import { Provider } from 'react-redux';

const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
      getParam: (name) => {
          if (name==='url') {
            return 'https://jestjs.io/';
          } 
          return null;
      }
    },
    ...props
  });

  describe('<HeaderRightElement />', () => {
    let props;
    let middlewares;
    let mockStore;
    
    beforeEach(() => {
        props = createTestProps({});
        middlewares = [thunk];
        mockStore = configureMockStore(middlewares);
      
      props.store = mockStore({
          uiElementsState: {
              showFilters: false,
              showPopUpMenu: false
          },
          articles: {
              filters: {}
          }
      });
    });
    
    test('renders the component', () => {
        const wrapper = shallow(<HeaderRightElement {...props} />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });

    test('Dispatches the correct action and payload, when filter are reset', () => {
        const { getAllByType } = render(<Provider store={props.store}><HeaderRightElement {...props} /></Provider>);
        fireEvent.press(getAllByType(TouchableOpacity)[0]);

        const expectedActions = [
            {
              payload: {
                  toggle: true
              },
              type: 'TOGGLE_POP_UP_MENU',
            },
          ];
      
        expect(props.store.getActions()).toEqual(expectedActions);
    });
      
});
