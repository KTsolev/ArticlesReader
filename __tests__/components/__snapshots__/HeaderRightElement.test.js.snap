// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderRightElement /> renders the component 1`] = `
<HeaderRightElement
  navigation={
    Object {
      "getParam": [Function],
      "navigate": [MockFunction],
    }
  }
  showPopUpMenu={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  togglePopup={[Function]}
/>
`;
